frontend/packages/pipelines-plugin/src/components/catalog/apis/artifactHub.ts

```diff
@@ -0,0 +1,144 @@
+import { ARTIFACTHUB_API_BASE_URL } from '../../../const';
+import { TaskModel, TaskModelV1Beta1 } from '../../../models';
+import { TektonTaskAnnotation } from '../../pipelines/const';
+import { ARTIFACTHUB } from '../../quicksearch/const';
+import { getInstalledFromAnnotation } from '../../quicksearch/pipeline-quicksearch-utils';
+import { ApiResult } from '../hooks/useApiResponse';
+
+export type ArtifactHubRepository = {
+  name: string;
+  kind: number;
+  url: string;
+  display_name: string;
+  repository_id: string;
+  organization_name: string;
+  organization_display_name: string;
+};
+
+export type ArtifactHubVersion = {
+  version: string;
+  contains_security_update: boolean;
+  prerelease: boolean;
+  ts: number;
+};
+
+export type ArtifactHubTask = {
+  package_id: string;
+  name: string;
+  description: string;
+  version: string;
+  display_name: string;
+  repository: ArtifactHubRepository;
+};
+
+export type ArtifactHubTaskDetails = {
+  package_id: string;
+  name: string;
+  description: string;
+  display_name: string;
+  keywords: string[];
+  platforms: string[];
+  version: ArtifactHubVersion[];
+  available_version: [];
+  content_url: string;
+  repository: ArtifactHubRepository;
+};
+
+const ARTIFACRHUB_TASKS_SEARCH_URL = `${ARTIFACTHUB_API_BASE_URL}/packages/search?offset=0&limit=60&facets=false&kind=7&deprecated=false&sort=relevance`;
+
+export const getArtifactHubTaskDetails = async (item: CatalogItem): Promise<any> => {
+  const API_BASE_URL = `${ARTIFACTHUB_API_BASE_URL}/packages/tekton-task`;
+  const { name, data } = item;
+  const {
+    task: {
+      version,
+      repository: { name: repoName },
+    },
+  } = data;
+  const url = `${API_BASE_URL}/${repoName}/${name}/${version}`;
+  return consoleProxyFetchJSON({ url, method: 'GET' });
+};
+
+export const useGetArtifactHubTasks = (hasPermission: boolean): ApiResult<any> => {
+  const [resultData, setResult] = React.useState([]);
+  const [loaded, setLoaded] = React.useState(false);
+  const [loadedError, setLoadedError] = React.useState<string>();
+
+  React.useEffect(() => {
+    let mounted = true;
+    if (hasPermission) {
+      consoleProxyFetchJSON<{ packages: ArtifactHubTask[] }>({
+        url: ARTIFACRHUB_TASKS_SEARCH_URL,
+        method: 'GET',
+      })
+        .then(({ packages }) => {
+          if (mounted) {
+            setLoaded(true);
+            setResult(packages);
+          }
+        })
+        .catch((err) => {
+          if (mounted) {
+            setLoaded(true);
+            setLoadedError(err?.message);
+          }
+        });
+    } else {
+      setLoaded(true);
+    }
+    return () => {
+      mounted = false;
+    };
+  }, [hasPermission]);
+  return [resultData, loaded, loadedError];
+};
+
+export const createArtifactHubTask = (url: string, namespace: string) => {
+  return consoleProxyFetchJSON({ url, method: 'GET' })
+    .then(async (task: K8sResourceKind) => {
```

Comment:
I think there is no need for async keyword here. I would prefer if all the code would use async-await on a top level. But if you prefer code with `.then()`, then you can just return the promise from k8sCreate.

```suggestion
    .then((task: K8sResourceKind) => {
```